From 4d682c1041f05f0fec4ecec7123d46290ac2002b Mon Sep 17 00:00:00 2001
From: "Wang, ArvinX" <arvinx.wang@intel.com>
Date: Fri, 1 Jun 2018 17:49:59 +0800
Subject: [PATCH 2/2] Fix RingtonePickerActivity crash while scrolling with
 keyboard

While scrolling with keyboard, it will play the fource item.
Due to the last item sometime can't play, need to ignore.

Change-Id: Ia3cdc0b0fe2ab3a084e908b23ca90d5b2c59a6d2
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-63302
Signed-off-by: Wang, ArvinX <arvinx.wang@intel.com>
---
 src/com/android/providers/media/RingtonePickerActivity.java | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/com/android/providers/media/RingtonePickerActivity.java b/src/com/android/providers/media/RingtonePickerActivity.java
index 7d91b6b..79c314c 100644
--- a/src/com/android/providers/media/RingtonePickerActivity.java
+++ b/src/com/android/providers/media/RingtonePickerActivity.java
@@ -493,6 +493,11 @@ public final class RingtonePickerActivity extends AlertActivity implements
      * On item selected via keys
      */
     public void onItemSelected(AdapterView parent, View view, int position, long id) {
+        // The "Add new ringtone" item was clicked.
+        if (position == mCursor.getCount() + mStaticItemCount) {
+            return;
+        }
+
         playRingtone(position, DELAY_MS_SELECTION_PLAYED);
 
         // In the buttonless (watch-only) version, preemptively set our result since we won't
-- 
1.9.1

